### PR TITLE
Add release native module article

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | Add release native module article
| Fixed ticket? |

### CODE REVIEW

File: content/news/posts/2023/release-native-module.md
Comment (matks) [inline content/news/posts/2023/release-native-module.md:32]: @Robin-Fischer-PS @florine2623 Please tell me if I write something wrong, I attempted to describe your work

File: content/news/posts/2023/release-native-module.md
Comment (kpodemski) [inline content/news/posts/2023/release-native-module.md:22]:
```suggestion
Each of these repositories hosts a `master` branch and a `dev` branch. The `dev` branch carries new changes for the incoming version, while the `master` branch holds the code that has been released. The latest commit on the `master` branch is the tag for the last release.
```

File: content/news/posts/2023/release-native-module.md
Comment (kpodemski) [inline content/news/posts/2023/release-native-module.md:24]:
```suggestion
As explained in [release process](https://www.prestashop-project.org/maintainers-guide/release-a-native-ps-module/), when conditions are met, the first release step is to open a Pull Request to merge commits from `dev` into `master`, just like [this one](https://github.com/PrestaShop/ps_facetedsearch/pull/646).
```

File: content/news/posts/2023/release-native-module.md
Comment (kpodemski) [inline content/news/posts/2023/release-native-module.md:30]:
```suggestion
Review must ensure the commits being transferred from one branch to another are the right ones (git usage errors can result in incorrect branch states). Reviewers will also verify the conditions (described in the release process) are met and that the version number qualifies given the changes carried out by the new version.
```

**Review by kpodemski — COMMENTED** [review]
I have a few comments, but I don't find them too relevant.

One thing I'm still uncertain of is that it looks like fine-tuned guidelines for project members to understand how it works. If you merge these two, that would be great content for the project members.

I wonder if the official blog is a good place for this, as I wonder if this article brings value to the community.

In my opinion, relevant information for the community could be:
- this information about testing a module in multiple versions and stressing the fact that it's time-consuming
- how often do releases happen
- how is it different between 1.7 and 8.0 regarding releases? If there's a module for 1.7 released, when will it be available in the back office?
- when and how it's decided to drop certain versions of PrestaShop core inside modules
- it would be worth reminding how to report bugs for modules or how to follow issues related to them

I wouldn't find many interesting things there if I were a regular community member.

**Comment by matks** [comment]
> that it looks like fine-tuned guidelines for project members to understand how it works. If you merge these two, that would be great content for the project members.

OK I can merge it with https://www.prestashop-project.org/maintainers-guide/release-a-native-ps-module/

**Comment by kpodemski** [comment]
Before you proceed, @matks, I'll ping @eternoendless and @coffrep who have a strategy for how the project should manage its blog.

File: content/news/posts/2023/release-native-module.md
Comment (coffrep) [inline content/news/posts/2023/release-native-module.md:30]:
```suggestion
Review must ensure the commit being transferred from one branch to another is the right one (git usage errors can result in incorrect branch states). Reviewers will also verify the conditions (described in the release process) are met and that the version number qualifies given the changes carried by the new version.
```

File: content/news/posts/2023/release-native-module.md
Comment (coffrep) [inline content/news/posts/2023/release-native-module.md:34]:
```suggestion
QA team members will fetch the code and proceed to testing. Module release tests are quite thorough.
```

File: content/news/posts/2023/release-native-module.md
Comment (coffrep) [inline content/news/posts/2023/release-native-module.md:36]:
```suggestion
The behavior of the new version will be tested on multiple PrestaShop versions, because a given module can be installed on multiple PrestaShop versions. For example [ps_facetedsearch release 3.8.0](https://github.com/PrestaShop/ps_facetedsearch/pull/646) can be installed on PrestaShop 1.7.6 and higher so tests were performed on multiple PrestaShop 1.7 versions.
```

File: content/news/posts/2023/release-native-module.md
Comment (coffrep) [inline content/news/posts/2023/release-native-module.md:38]:
```suggestion
Secondly, the QA team also verifies the module upgrade happens smoothly. Modules have the ability to run specific scripts when updated, for example to enable or disable hooks or to update a given SQL table. So multiple use cases are tested: fresh installs of the new module or updates of an existing module in a lower version.
```

File: content/news/posts/2023/release-native-module.md
Comment (coffrep) [inline content/news/posts/2023/release-native-module.md:40]:
```suggestion
This test campaign might detect blocking bugs that prevent the release from happening: a new version of a module should be an improvement of a previous version, not a regression. Blocking bugs must be fixed by submitting pull requests against the `dev` branch, until the `dev` branch is acknowledged free of blocking bugs.
```

File: content/news/posts/2023/release-native-module.md
Comment (coffrep) [inline content/news/posts/2023/release-native-module.md:46]:
```suggestion
When the release pull request is merged, the last steps of the [release process](https://www.prestashop-project.org/maintainers-guide/release-a-native-ps-module/) can be performed to complete the process: create a GitHub release that will be picked up by [PrestaShop Modules Distribution API](https://github.com/PrestaShop/distribution-api) which is polled by PrestaShop instances for new updates, downloads, and proceed to GitHub milestones as well as release curation.
```

File: content/news/posts/2023/release-native-module.md
Comment (coffrep) [inline content/news/posts/2023/release-native-module.md:24]:
```suggestion
So as explained in [release process](https://www.prestashop-project.org/maintainers-guide/release-a-native-ps-module/), when conditions are met, the first release step is to open a pull request to merge commits from `dev` into `master`, just like [this one](https://github.com/PrestaShop/ps_facetedsearch/pull/646).
```

File: content/news/posts/2023/release-native-module.md
Comment (coffrep) [inline content/news/posts/2023/release-native-module.md:28]:
```suggestion
This pull request that aims to merge commits from `dev` into `master` must be reviewed and tested.
```

File: content/news/posts/2023/release-native-module.md
Comment (coffrep) [reply content/news/posts/2023/release-native-module.md:30]: Hi Kpodemski, this version is good if "commits" is plural in the text - please do not take into account my suggestion here if commits needs to be plural

**Review by coffrep — CHANGES_REQUESTED** [review]
Hi team, I've added a few suggestions on the post, do feel free to approve changes or let me know if they don't make sense.

File: content/news/posts/2023/release-native-module.md
Comment (coffrep) [reply content/news/posts/2023/release-native-module.md:24]: For your info, on the GitHub site and their documentation, they write pull request in lower case. Just FYI...

**Review by eternoendless — COMMENTED** [review]
This looks like a process and not "news". Why not improve this page instead? https://www.prestashop-project.org/maintainers-guide/release-a-native-ps-module/

**Pull request closed without merging**
